akka-http-core/src/main/scala/akka/http/impl/util/Java6Compat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.util
+
+import java.lang.reflect.{ InvocationTargetException, Method }
+import javax.net.ssl.SSLParameters
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Enables accessing SslParameters even if compiled against Java 6.
+ */
+private[http] object Java6Compat {
```

Comment:
Having thought once more about it (and after searching for mentions of "getHostString") I think that we should even move `getHostStringJava6Compatible` to `akka.io` or `akka.util` as there are a lot of `TODO`s all over the place.